packages/edit-site/src/store/actions.js

```diff
@@ -236,3 +242,68 @@ export function updateSettings( settings ) {
 		settings,
 	};
 }
+
+/**
+ * Reverts a template to its original theme-provided file.
+ *
+ * @param {Object} template The template to revert.
+ */
+export function* revertTemplate( template ) {
+	if ( ! isTemplateRevertable( template ) ) {
+		yield controls.dispatch(
+			'core/notices',
+			'createErrorNotice',
+			__( 'This template is not revertable.' ),
+			{ type: 'snackbar' }
+		);
+		return;
+	}
+
+	try {
+		yield controls.dispatch(
+			'core',
+			'deleteEntityRecord',
+			'postType',
+			'wp_template',
+			template.id
+		);
+
+		const fileTemplate = yield controls.resolveSelect(
+			'core',
+			'getEntityRecord',
+			'postType',
+			'wp_template',
+			template.id
+		);
```

Comment:
This doesn't seem to work.
No REST requests are fired, and it always returns `undefined`.

On the other hand, if I try to get _all_ entity records first, this will return the correct one.
E.g. adding this before:
```js
const allTemplates = yield controls.resolveSelect(
  'core',
  'getEntityRecords',
  'postType',
  'wp_template'
);
```

Reply:
Looks like the entity resolution isn't being invalidated.

If you add the snippet below, before `resolveSelect`, then it will work just fine.
```js
yield controls.dispatch(
	'core',
	'invalidateResolution',
	'getEntityRecord',
	[ 'postType', 'wp_template', template.id ]
);
```

Now the question is, why it isn't invalidated after delete? It **does** invalidate the `getEntityRecords( 'postType', 'wp_template' )` though. That's why adding the snippet you mentioned in the comment works.

Reply:
What happens if you had unsaved edits? it seems we should also reset the edits right?